### PR TITLE
Fixes a bug turning weak scuttlebots into strong ones.

### PR DESCRIPTION
**Pull request by Bartimeus973**
[bug]

## About the PR 
Fixes an issue that lets you turn weak scuttlebots like the det's into strong ones.

## Why's this needed? 
No free antag items.

### CODE REVIEW

File: code/obj/item/clothing/glasses.dm [hunk 1]
```diff
@@ -421,24 +421,23 @@
 		else
 			boutput(user, "<span class='alert'>You put on the glasses but they show no signal. The scuttlebot is likely destroyed.</span>")
 
-	attack(obj/item/W, mob/M)
```
Comment (ZeWaka) [inline code/obj/item/clothing/glasses.dm:424]: why is it a living mob

Comment (Bartimeus973) [reply code/obj/item/clothing/glasses.dm:424]: Its the scuttlebot, i'll revert it

File: code/obj/item/clothing/glasses.dm [hunk 2]
```diff
@@ -423,22 +423,21 @@
 
 	attack(obj/item/W, mob/M)
 		if(istype(W, /mob/living/critter/robotic/scuttlebot))
```
Comment (ZeWaka) [inline code/obj/item/clothing/glasses.dm:425]: how is an item a mob

Comment (Bartimeus973) [reply code/obj/item/clothing/glasses.dm]: Updated. Hopefully i stopped being dumb.

File: code/obj/item/clothing/glasses.dm [hunk 3]
```diff
@@ -421,29 +421,25 @@
...
-				var/mob/living/critter/robotic/scuttlebot/weak/O = S
-				if (O.linked_hat != null)
-					O.linked_hat.set_loc(get_turf(O))
-				else
-					new /obj/item/clothing/head/det_hat/gadget(get_turf(src))
-				boutput(M, "You stuff the goggles back into the detgadget hat. It powers down with a low whirr.")
-				qdel(O)
-				qdel(src)
+	attack(mob/living/critter/robotic/scuttlebot/W, mob/M)
+		if (connected_scuttlebot != W)
```
Comment (ZeWaka) [inline code/obj/item/clothing/glasses.dm:425]: this is still incorrect. You are not guaranteed to be passed a mob. Look at the proc you are overriding.

**Pull request merged**